### ISSUE TITLE
libdoc needs format as argument in RF4.0a2

### ISSUE THREAD

**Issue by AHm74** [robotframework/robotframework#3756]
Environment python 3.7, win10

In RF 3.1.1, 3.2.2 and 4.0a1 the following works. But when playing with RF4.0a2 I'm not able to generate documentation without specifying format. 
![image](https://user-images.githubusercontent.com/4309313/98128248-f4cf3200-1eb7-11eb-9da1-9a6b61cca816.png)

![image](https://user-images.githubusercontent.com/4309313/98127680-44612e00-1eb7-11eb-93e7-0454405ba86f.png)

**Comment by pekkaklarck** [comment]
Thanks for testing alphas and reporting this issue! The problem was due to a refactoring that moved option validation to place where it wasn't done when Libdoc was used programmatically like this. Part of the validation was setting the default value for the format based on the output file extension and because that wasn't done format was `None`. This was fixed by the commit above and I added also unit tests to for this kind of usage.
